docs/integrations/databases/supabase.mdx

```diff
@@ -3,170 +3,666 @@ title: Integrate Supabase with Clerk
+        process.env.NEXT_PUBLIC_SUPABASE_KEY ?? "",
+        {
+          global: { 
+            headers: { 
+              Authorization:c `Bearer ${supabaseAccessToken}`
+            }
+          },
+      });
+      return supabase;
+    };
+    ```
+    ```ts filename="components/supabase.tsx"
+    import { useAuth } from '@clerk/clerk-react';
+    import { createClient } from "@supabase/supabase-js";
+    
+    // Uses the JWT passed to it to initialize the client object.
+    // This JWT will come from your JWT template, and be sent
+    // with all requests to supabase.
+    const supabaseClient = async (supabaseAccessToken: string) => {
+      const supabase = createClient(
+        process.env.REACT_APP_SUPABASE_URL ?? "",
+        process.env.REACT_APP_SUPABASE_KEY ?? "",
+        {
+          global: {
+            headers: {
+              Authorization: `Bearer ${supabaseAccessToken}`,
+            },
+          },
+        }
+      );
+      return supabase;
+    };
+    ```
+    </CodeBlockTabs>
+1. Next, define your component. It will:
+    - Call `useAuth()` to get `userId` and `getToken()`.
+    - Define a function that gets a Supabase client using an access token from your JWT template.
+    <CodeBlockTabs type="framework" options={["Next.js App Router", "Next.js Pages Router", "React"]}>
+    ```tsx filename="app/supabase/page.tsx"
+    export default function Supabase() {
+      const { getToken, userId } = useAuth();
+
+      const getSupabaseClient = async () => {
+        // Replace with your JWT template name
+        const supabaseAccessToken = await getToken({ template: "supabase" });
+        if (!supabaseAccessToken)
+          throw new Error("Failed to get token from JWT template");
+        return await supabaseClient(supabaseAccessToken);
+      };
+      return null;
+    }
+    ```
+    ```tsx filename="pages/supabase/index.tsx"
+    export default function Supabase() {
+      const { getToken, userId } = useAuth();
+
+      const getSupabaseClient = async () => {
+        // Replace with your JWT template name
+        const supabaseAccessToken = await getToken({ template: "supabase" });
+        if (!supabaseAccessToken)
+          throw new Error("Failed to get token from JWT template");
+        return await supabaseClient(supabaseAccessToken);
+      };
+      return null;
+    }
+    ```
+    ```tsx filename="components/supabase.tsx"
+    export default function Supabase() {
+      const { getToken, userId } = useAuth();
+
+      const getSupabaseClient = async () => {
+        // Replace with your JWT template name
+        const supabaseAccessToken = await getToken({ template: "supabase" });
+        if (!supabaseAccessToken)
+          throw new Error("Failed to get token from JWT template");
+        return await supabaseClient(supabaseAccessToken);
+      };
+      return null;
+    }
+    ```
+    </CodeBlockTabs>
+1. Next, add code to your component to send and get posts from your database. 
```

Comment:
In your App Router example, you're utilizing client code in a component that isn't explicitly using "use client" which will break if we don't make the specific callout. 

App router components are "Server first" in that, not marking them explicitly to "use client" will break when using hooks like `useRef`, `useState`, etc

Reply:
Yeah, I have `"use client"` at the top of the file in the first snippet, and at the top of the full snippet in the final snippet

Reply:
I see what he's doing - each step is building on the same /app/supabase/page.tsx - so that you get to the final step, which shows u what the code should look like in its entirety